client/landing/stepper/declarative-flow/internals/steps-repository/free-setup/index.tsx

```diff
@@ -52,36 +162,34 @@ const FreeSetup: Step = function FreeSetup( { navigation } ) {
 		setTagline( site.description );
 	}, [ site ] );
 
-	const handleSubmit = async ( event: FormEvent ) => {
-		event.preventDefault();
-		setInvalidSiteTitle( ! siteTitle.trim().length );
-
-		setSiteDescription( tagline );
-		setSiteTitle( siteTitle );
-
-		if ( selectedFile && base64Image ) {
-			try {
-				setSiteLogo( base64Image );
-			} catch ( _error ) {
-				// communicate the error to the user
-			}
+	useEffect( () => {
+		if ( siteTitle.trim().length > 0 && ! showGeneratedDomainFormField ) {
+			// Debounce for random domain generation API while user is typing site title
+			debounceGenerateRandomDomainAPI( siteTitle );
 		}
+	}, [ siteTitle, showGeneratedDomainFormField, debounceGenerateRandomDomainAPI ] );
 
-		if ( siteTitle.trim().length ) {
-			submit?.( { siteTitle, tagline } );
+	// Splits generated domain suggestion for styling purposes
+	useEffect( () => {
+		if ( generatedDomainSuggestion ) {
```

Comment:
This is for styling the domain name similar to the Launchpad sidebar URL.

Domain: BOLD font weight
TLD: normal font weight

Reply:
`NB`: Should we make a shared, custom hook for this logic?